lib/trip_dispatcher.rb

```diff
@@ -19,6 +21,45 @@ def find_passenger(id)
       return @passengers.find { |passenger| passenger.id == id }
     end
 
+    def find_driver(id)
+      Driver.validate_id(id)
+      return @drivers.find { |driver| driver.id == id }
+    end
+
+    def driver_assigned
+      driver = @drivers.find { |driver| driver.status == :AVAILABLE }
+      return driver
+    end
+
+    def request_trip(passenger_id)
+      driver = driver_assigned
+      return nil if driver == nil
+
+      start_time = Time.now
+      end_time = nil
+      passenger = find_passenger(passenger_id)
+      trip_data = {
+          id: @trips.length + 1,
+          passenger: passenger,
+          start_time: start_time,
+          end_time: end_time,
+          cost: nil,
+          rating: nil,
+          driver_id: driver.id,
+          driver: driver
+      }
+
+      new_trip_instance = RideShare::Trip.new(trip_data)
```

Comment:
You got a warning for this.  To fix it directly use keyword arguments instead of passing in a hash:

```suggestion
      new_trip_instance = RideShare::Trip.new(
        id: @trips.length + 1,
        passenger: passenger,
        start_time: start_time,
        end_time: end_time,
        cost: nil,
        rating: nil,
        driver_id: driver.id,
        driver: driver
      )
```